frontend/webpack.dev.js

```diff
@@ -0,0 +1,27 @@
+const { merge } = require('webpack-merge');
+const common = require('./webpack.common.js');
+
+module.exports = merge(common, {
+    mode: 'development',
+    // devtool from here https://github.com/TypeStrong/ts-loader
+    devtool: 'inline-source-map',
+    devServer: {
+        contentBase: './dist',
+        host: '0.0.0.0',
+        port: 9000,
+
+        hot: true,
+        // To fallback on index.html when page not found (needed for client side routing).
+        historyApiFallback: true,
+
+        // Redirect api requests to the backend
+        proxy: {
+            '/api': {
+                target: 'http://localhost:8080',
```

Comment:
Btw, what localhost is this? Is it the machine where the server is running or the client (frontend)? By hardcoding `localhost` here do we require user to run server and browser on the same machine?

Reply:
first of all, I need to notice that this is only dev config, so it's only used when you run code locally.

In production/staging frontend, backend and ngnix will be at the under the same host and this config is not be used. All api requests in the code will like /api/blabla and will be addressed to the host/IP where the website is running.